src/processor.cpp

```diff
@@ -0,0 +1,401 @@
+                    // This will add items to a space near the processor, if it's flagged as NOITEM.
+                    item_it = items.erase( item_it );
+                } else {
+                    item_it++;
+                    processed_is_present = true;
+                }
+            }
+            //Option 2
+            processed_is_present = true;
+            for( auto i : items ) {
+                if( processed_materials.find( i->id ) == container.end() && i.id  != processor.fuel_type ) {
+                    add_msg( m_bad, _("You find some %s, which can't be processed here."), i.tname( 1, false ).c_str() );
+                    add_msg( _("You must remove it before you proceed.") );
+                    return;
+                }
+            }
+        //TODO: If there's nothing on the tile, check player inventory
+        // to see if they have anything to process
+        if( !processed_is_present ){
+            add_msg(m_info, _("There's nothing that can be processed."));
+            return;
+        }
+        //  At this point, we should have only usable reagents on the tile,
+        //  Currently in a very unordered fashion (keg code could help)
+        //  This next loop is currently not used, could help sort items into a
+        //  predefined order (based on order in process.json maybe?)
+        std::vector<item> materials;
+        std::vector<integer> charges;
+        item &fuel;
+        items = g->m.i_at( examp );
+        for( auto item_it = items.begin(); item_it != items.end(); ) {
+            if (item_it.typeId() == processor.fuel_type ){
+                fuel = item_it;
+                continue;
+            }
+            for( auto component_it = processed_materials.begin(); component_it != processed_materials.end(); ){
+                //If we found which components it belongs to
+                if( item_it.typeId() == component_it.typeId() ) {
+                    materials.push_back( item_it.typeId() );
+                    charges.push_back( item_it.charges );
+                    break;
+                }
+                component_it++;
+            }
+            item_it++;
+        }
+        //  Now we know how many charges of each components do we have, along with how much fuel
+        //  Current implementation plan is to assume use of one on each components and one fuel_intake number of fuel charges
+        //  per charge of output produced
+
+        //  Should it be volume based?
+        //TODO: calculate skill gain
+        //TODO: calculate output here, or in second phase, based on charges and fuel present at start
+
+        //TODO: make it skill use dependent? Would be silly to ask for fire on a CNC machine down the road
+        if( !p.has_charges( "fire" , 1 ) ) {
+            add_msg( _("This process is ready to be started, but you have no fire source.") );
+            return;
+        } else if( !query_yn( _( processor.full_message ) ) ) {
```

Comment:
`full_message` is a C++ `std::string`, but `_()` function only accepts C-style `const char*` string. You have to feed it `processor.full_message.c_str()`